### PR TITLE
DOC: More details about creating Azure resources

### PR DESCRIPTION
**Pull request by ant0nsc**
Extending the AzureML onboarding instructions, including a fully script-driven setup.

### CODE REVIEW

**Review by kenza-bouzid — APPROVED** [review]
LGTM!

File: docs/source/azure_setup.md [hunk 1]
```diff
@@ -1,13 +1,109 @@
...
+
+There are two ways to set up all necessary resources, either via the Azure portal or via the Azure Commandline Interface (CLI).
+We recommend the CLI because all necessary resources can be easily created via a single script.
+
+## Creating an AzureML workspace via the Azure Portal
+
+If you prefer to create your workspace via the web UI on the Azure Portal, please follow the steps below.
 
 - [Create a resource
   group](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/manage-resource-groups-portal)
```
Comment (peterhessey) [inline docs/source/azure_setup.md:18]:
```suggestion
  group.](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/manage-resource-groups-portal)
```

File: docs/source/azure_setup.md [hunk 2]
```diff
@@ -1,13 +1,109 @@
...
 - [Create a storage account for
   datasets](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create?tabs=azure-portal)
```
Comment (peterhessey) [inline docs/source/azure_setup.md:20]:
```suggestion
  datasets.](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create?tabs=azure-portal)
```

File: docs/source/azure_setup.md [hunk 3]
```diff
@@ -1,13 +1,109 @@
...
 - [Create an AzureML workspace, compute instances and compute clusters](https://docs.microsoft.com/en-us/azure/machine-learning/quickstart-create-resources)
```
Comment (peterhessey) [inline docs/source/azure_setup.md:21]:
```suggestion
- [Create an AzureML workspace, compute instances and compute clusters.](https://docs.microsoft.com/en-us/azure/machine-learning/quickstart-create-resources)
```

File: docs/source/azure_setup.md [hunk 4]
```diff
@@ -1,13 +1,109 @@
...
 
+## Creating an AzureML workspace via the Azure Commandline Tools
+
+A pureley commandline-drive setup is possible via the [Azure Commandline Tools](https://docs.microsoft.com/en-us/cli/azure/). These tools are available for multiple platforms, including Linux, Mac, and Windows.
```
Comment (peterhessey) [inline docs/source/azure_setup.md:25]:
```suggestion
A purely command line-drive setup is possible via the [Azure Command Line Tools](https://docs.microsoft.com/en-us/cli/azure/). These tools are available for multiple platforms, including Linux, Mac, and Windows.
```

File: docs/source/azure_setup.md [hunk 5]
```diff
@@ -1,13 +1,109 @@
...
+
+After downloading the commandline tools, you can run the following command add the `ml` extension that is required to create an AzureML workspace:
```
Comment (peterhessey) [inline docs/source/azure_setup.md:27]:
```suggestion
After downloading the commandline tools, you can run the following command to add the `ml` extension that is required to create an AzureML workspace:
```

File: docs/source/azure_setup.md [hunk 6]
```diff
@@ -1,13 +1,109 @@
...
+
+```bash
+az extension add --name ml
+```
+
+### Documentation
+
+- [Creating AzureML workspaces via the CLI](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-manage-workspace-cli)
```
Comment (peterhessey) [inline docs/source/azure_setup.md:35]:
```suggestion
- [Creating AzureML workspaces via the CLI.](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-manage-workspace-cli)
```

File: docs/source/azure_setup.md [hunk 7]
```diff
@@ -1,13 +1,109 @@
...
+- [Creating Azure storage accounts via the CLI](https://docs.microsoft.com/en-us/cli/azure/storage?view=azure-cli-latest)
```
Comment (peterhessey) [inline docs/source/azure_setup.md:36]:
```suggestion
- [Creating Azure storage accounts via the CLI.](https://docs.microsoft.com/en-us/cli/azure/storage?view=azure-cli-latest)
```

File: docs/source/azure_setup.md [hunk 8]
```diff
@@ -1,13 +1,109 @@
...
+- [Schema for creating datastores](https://docs.microsoft.com/en-us/azure/machine-learning/reference-yaml-datastore-blob)
```
Comment (peterhessey) [inline docs/source/azure_setup.md:37]:
```suggestion
- [Schema for creating datastores.](https://docs.microsoft.com/en-us/azure/machine-learning/reference-yaml-datastore-blob)
```

File: docs/source/azure_setup.md [hunk 9]
```diff
@@ -1,13 +1,109 @@
...
+- [Creating datastores via the CLI](https://docs.microsoft.com/en-us/cli/azure/ml/datastore?view=azure-cli-latest)
+- [Create a shared access signature](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-user-delegation-sas-create-cli)
+
+### Collecting the necessary information
+
+Find out which Azure datacenter locations you can use:
```
Comment (peterhessey) [inline docs/source/azure_setup.md:43]:
```suggestion
Find out which Azure data centre locations you can use:
```

File: docs/source/azure_setup.md [hunk 10]
```diff
@@ -1,13 +1,109 @@
...
+az ml datastore create --file ${datastorefile} --resource-group ${prefix}rg --workspace-name ${prefix}
+rm ${datastorefile}
+```
+
+Note that the datastore will use the storage account key to authenticate. You may want to switch that to Shared Access Signature (SAS).
+
+Now that you have created the core AzureML workspace, you need to
+[create a compute cluster](https://docs.microsoft.com/en-us/azure/machine-learning/quickstart-create-resources#cluster).
+
+The final step is to download the workspace configuration file, as descreibed [here](#accessing-the-workspace).
```
Comment (peterhessey) [inline docs/source/azure_setup.md:100]:
```suggestion
The final step is to download the workspace configuration file, as described [here](#accessing-the-workspace).
```

**Review by peterhessey — CHANGES_REQUESTED** [review]
Minor spelling and formatting changes requests

**Review by peterhessey — APPROVED** [review]

**Pull request merged**